src/wms/src/main/java/org/geoserver/wms/capabilities/Capabilities_1_3_0_Transformer.java

```diff
@@ -1363,18 +1363,19 @@ protected void handleAttribution(PublishedInfo layer) {
          * Writes layer LegendURL pointing to the user supplied icon URL, if any, or to the proper
          * GetLegendGraphic operation if an URL was not supplied.
          *
-         * <p>It is common practice to supply a URL to a WMS accesible legend graphic when it is
+         * <p>It is common practice to supply a URL to a WMS accessible legend graphic when it is
          * difficult to create a dynamic legend for a layer.
          *
          * @param layer The layer.
          * @param legend The user specified legend url. If null a default url pointing back to the
          *     GetLegendGraphic operation will be automatically created.
-         * @param style The styel for the layer.
-         * @param sampleStyle The styel to use for sample sizing.
+         * @param style The style for the layer.
+         * @param sampleStyle The style to use for sample sizing.
          */
         protected void handleLegendURL(
                 LayerInfo layer, LegendInfo legend, StyleInfo style, StyleInfo sampleStyle) {
-            if (legend != null) {
+            // if legend is valid, use it
+            if (validateLegendInfo(legend)) {
```

Comment:
Looks good to me. However, the very same fix also needs to be done in the WMS 1.1 caps transformer, can you add that one too and the associated test?

Reply:
Added WMS 1.1.1 fix and test as well, thanks!